packages/flutter/lib/src/widgets/nested_scroll_view.dart

```diff
@@ -190,7 +190,8 @@ class NestedScrollView extends StatefulWidget {
     @required this.headerSliverBuilder,
     @required this.body,
     this.dragStartBehavior = DragStartBehavior.start,
-  }) : assert(scrollDirection != null),
+  }) : coordinator = NestedScrollCoordinator(),
```

Comment:
This can't be on a widget. You can reuse the same widget in multiple places in the tree, for one.